OpenWrt devel install - sysrepoctl

Hi,
I got an error when trying to install sysrepo from devel branch for OpenWrt.

`Install the project...
-- Install configuration: "release"
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo/yang/
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo/data/
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo/data/internal/
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/var/run/sysrepo-subscriptions
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/usr/lib/sysrepo/plugins
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo/yang/internal/sysrepo-persistent-data.yang
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo/yang/internal/sysrepo-module-dependencies.yang
-- Installing: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/ipkg-install/etc/sysrepo/yang/internal/sysrepo-notification-store.yang
-- Exec: /home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/src/sysrepoctl --install --yang=/home/nagy/openWRT/chaos_calmer/openwrt/build_dir/target-i386_pentium4_musl-1.1.16/sysrepo-be8d146bc4c580e9a8914fa94ef1729d1467122b/yang/ietf-netconf-acm@2012-02-22.yang --permissions=644
CMake Error at cmake_install.cmake:127 (message):
  Error: No such file or directory`

Is there any other solution how to do "sysrepoctl --install ..." functionality manually? I think "sysrepoctl" binary is compiled for OpenWrt and I can't run it on my machine.
Thanks.

Regards,
Peter

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,9 @@ endif()
 set(CMAKE_C_FLAGS_RELEASE "-DNDEBUG -O2")
 set(CMAKE_C_FLAGS_DEBUG   "-g -O0")
 
+set(CALL_TARGET_BINS_DIRECTLY ON CACHE BOOL "Enable calling sysrepoctl/sysrepocfg directly at install time")
+set(INDIRECT_YANG_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/yang/" CACHE PATH "Where to store YANG files when not executing sysrepoctl directly")
+
 if(NOT UNIX)
     message(FATAL_ERROR "Only Unix-like systems are supported.")
 endif()
@@ -192,8 +195,13 @@ macro(EXEC_AT_INSTALL_TIME CMD)
         )
 endmacro(EXEC_AT_INSTALL_TIME)
 
+set(SH_INSTALL_YANG_CMDS "#!/bin/bash\n\nset -eux -o pipefail\nshopt -s failglob\n\nSYSREPOCTL=sysrepoctl\n\n")
 macro(INSTALL_YANG MODULE_NAME REVISION PERMISSIONS)
-    EXEC_AT_INSTALL_TIME("${CMAKE_BINARY_DIR}/src/sysrepoctl --install --yang=${CMAKE_CURRENT_SOURCE_DIR}/yang/${MODULE_NAME}${REVISION}.yang --permissions=${PERMISSIONS}")
+    if(CALL_TARGET_BINS_DIRECTLY)
+        EXEC_AT_INSTALL_TIME("${CMAKE_BINARY_DIR}/src/sysrepoctl --install --yang=${CMAKE_CURRENT_SOURCE_DIR}/yang/${MODULE_NAME}${REVISION}.yang --permissions=${PERMISSIONS}")
+    else()
+        set(SH_INSTALL_YANG_CMDS "${SH_INSTALL_YANG_CMDS}\\\${SYSREPOCTL} --install --yang=${INDIRECT_YANG_INSTALL_DIR}/${MODULE_NAME}${REVISION}.yang --permissions=${PERMISSIONS}\n")
+    endif()
 endmacro(INSTALL_YANG)
 
 # Examples
@@ -296,4 +304,13 @@ INSTALL_YANG("nc-notifications" "" "666")
 
 if(GEN_LANGUAGE_BINDINGS)
     add_subdirectory(swig)
+endif()
+
+if(NOT CALL_TARGET_BINS_DIRECTLY)
+    install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/yang/"
+            DESTINATION "${INDIRECT_YANG_INSTALL_DIR}"
+            FILES_MATCHING PATTERN "*.yang")
+    set(install_yang_file_name "${CMAKE_CURRENT_BINARY_DIR}/install-yang.sh")
+    install(CODE "file(WRITE \"${install_yang_file_name}\" \"${SH_INSTALL_YANG_CMDS}\")")
+    install(CODE "message(STATUS \"Installation commands were written to ${install_yang_file_name}\")")
 endif()

